sycl/include/sycl/types.hpp

```diff
@@ -1544,8 +1544,16 @@ template <typename VecT, typename OperationLeftT, typename OperationRightT,
           template <typename> class OperationCurrentT, int... Indexes>
 class SwizzleOp {
   using DataT = typename VecT::element_type;
-  using CommonDataT = std::common_type_t<typename OperationLeftT::DataT,
-                                         typename OperationRightT::DataT>;
+  using CommonDataT = std::conditional_t<
```

Comment:
I'd add `using OpLeftDataT`/`OpRightDataT` here, maybe with different names. Also, we need a comment explaing what we're doing and **why**.